pkgs/development/compilers/rust/default.nix

```diff
@@ -19,17 +19,17 @@
     }.${cpu.name} or cpu.name;
   in "${cpu_}-${vendor.name}-${kernel.name}${lib.optionalString (abi.name != "unknown") "-${abi.name}"}";
 
-  makeRustPlatform = { rustc, cargo, ... }: {
+  makeRustPlatform = { rustc, cargo, ... }: rec {
     rust = {
       inherit rustc cargo;
     };
 
-    buildRustPackage = callPackage ../../../build-support/rust {
-      inherit rustc cargo;
+    fetchcargo = buildPackages.callPackage ../../../build-support/rust/fetchcargo.nix {
+      inherit cargo;
+    };
 
-      fetchcargo = buildPackages.callPackage ../../../build-support/rust/fetchcargo.nix {
-        inherit cargo;
-      };
+    buildRustPackage = callPackage ../../../build-support/rust {
+      inherit rustc cargo fetchcargo;
```

Comment:
How this passed `fetchcargo` get used?

Reply:
Not sure I understand? It’s used here:

https://github.com/NixOS/nixpkgs/blob/83ac9c07e4df352e177ebdf978320089a137183a/pkgs/build-support/rust/default.nix#L31

Reply:
Oh my bad, I looked at the changes, that you add `fetchcargo` argument but do not add any uses of it. Now I see that the point is to pass the local version referencing correct `cargo`.